test/TestHelper.cpp

```diff
@@ -547,26 +545,21 @@ bytes importData(json_spirit::mObject const& _o)
 
 std::string compileLLL(std::string const& _code)
 {
-	FILE *fp;
-	char path[1035];
-	FILE *ft = fopen("temp.sol", "w");
-	if (ft == NULL)
-		cerr << "Error creating temp file for lllc";
-	else
-		fputs(_code.c_str(), ft);
-	fclose(ft);
-
-	fp = popen("../../solidity/lllc/lllc temp.sol", "r");
+	char input[1024];
+	boost::filesystem::path path = boost::filesystem::unique_path();
+	std::string filename = path.string() + "/code.sol";
+	std::string lllc ("../../solidity/lllc/lllc ");
+	lllc += filename.c_str();
+	writeFile(filename, _code);
+
+	FILE *fp = popen(lllc.c_str(), "r");
 	if (fp == NULL)
-		cerr << "Failed to run lllc";
-
-	fgets(path, sizeof(path)-1, fp);
-
+		BOOST_ERROR("Failed to run lllc");
+	fgets(input, sizeof(input)-1, fp);
 	pclose(fp);
-	if(remove("temp.sol") != 0)
-		cerr << "Error deleting temp file for lllc";
 
-	string result(path);
+	boost::filesystem::remove_all(path);
+	string result(input);
```

Comment:
`return string("0x") + boost::trim_copy(result);` - check that trim_copy also removes `\n` because I think this is what you want to remove, right?